modules/cluster/security_groups.tf

```diff
@@ -69,26 +71,38 @@ resource "aws_security_group_rule" "node_ingress_self" {
   type                     = "ingress"
 }
 
-resource "aws_security_group_rule" "node_ingress_cluster" {
+resource "aws_security_group_rule" "node_ingress_cluster_sg" {
   count = var.legacy_security_groups ? 1 : 0
 
-  description              = "Allow worker Kubelets and pods to receive communication from the cluster control plane"
+  description              = "Allow nodes to communicate with nodes in the cluster security group"
   from_port                = 0
   to_port                  = 65535
-  protocol                 = "tcp"
+  protocol                 = "-1"
+  security_group_id        = aws_eks_cluster.control_plane.vpc_config.0.cluster_security_group_id
+  source_security_group_id = aws_security_group.node[count.index].id
+  type                     = "ingress"
+}
+
+resource "aws_security_group_rule" "cluster_sg_ingress_node" {
+  count = var.legacy_security_groups ? 1 : 0
+
+  description              = "Allow nodes to communicate with nodes in the cluster security group"
+  from_port                = 0
+  to_port                  = 65535
+  protocol                 = "-1"
   security_group_id        = aws_security_group.node[count.index].id
-  source_security_group_id = aws_security_group.control_plane[count.index].id
+  source_security_group_id = aws_eks_cluster.control_plane.vpc_config.0.cluster_security_group_id
   type                     = "ingress"
 }
 
-resource "aws_security_group_rule" "cluster_ingress_node_https" {
+resource "aws_security_group_rule" "node_ingress_cluster" {
   count = var.legacy_security_groups ? 1 : 0
 
-  description              = "Allow pods to communicate with the cluster API Server"
-  from_port                = 443
-  to_port                  = 443
+  description              = "Allow worker Kubelets and pods to receive communication from the cluster control plane"
+  from_port                = 0
+  to_port                  = 65535
   protocol                 = "tcp"
-  security_group_id        = aws_security_group.control_plane[count.index].id
-  source_security_group_id = aws_security_group.node[count.index].id
+  security_group_id        = aws_security_group.node[count.index].id
+  source_security_group_id = aws_eks_cluster.control_plane.vpc_config.0.cluster_security_group_id
   type                     = "ingress"
 }
```

Comment:
Aren't these two resources redundant? They both enable comm from the control plane to the nodes and the first allows all protocols instead of just TCP. It feels like the second one should be removed. Am I missing something? Thanks. :bow:

Reply:
yes you are correct.